### PR TITLE
Do not emit events in mirror mode.

### PR DESCRIPTION
**Pull request by russjones**
**Description**

Update mirror mode (for both the memory and SQLite backends) to no longer emit events when an element expires. This allows caches to handle update/delete logic themselves.

This fixes an issue where services.ProxyWatcher was not getting updates to the list of proxies.

### CODE REVIEW

File: lib/services/proxywatcher.go [hunk 1]
```diff
@@ -127,15 +128,24 @@ func (cfg *ProxyWatcherConfig) CheckAndSetDefaults() error {
...
+	p.mu.RLock()
+	defer p.mu.RUnlock()
 	return p.current
 }
 
 // setCurrent sets currently active proxy list
 func (p *ProxyWatcher) setCurrent(proxies []Server) {
-	p.Lock()
-	defer p.Unlock()
+	p.mu.Lock()
```
Comment (klizhentas) [inline lib/services/proxywatcher.go]: why this change?

File: lib/services/proxywatcher.go [hunk 2]
```diff
@@ -215,7 +225,7 @@ func (p *ProxyWatcher) watch() error {
 	case <-watcher.Done():
 		return trace.ConnectionProblem(watcher.Error(), "watcher is closed")
 	case <-reloadC:
-		p.Debugf("Triggering scheduled reload.")
+		p.log.Debugf("Triggering scheduled reload.")
```
Comment (klizhentas) [inline lib/services/proxywatcher.go]: why this change?

**Review by klizhentas — COMMENTED** [review]
Looks good except a couple of refactoring style that is inconsistent with the rest of the codebase

File: lib/services/proxywatcher.go (hunk 2 above)
Comment (russjones) [reply lib/services/proxywatcher.go]: Embedding `sync.RWMutex` and `logrus.Entry` trashes autocomplete for me.

**Review by klizhentas — APPROVED** [review]
